### ISSUE TITLE
[HeaderAction and SuiteHeader] Fix accessibility scan violations

### ISSUE THREAD

**Issue by kalyanixraut** [carbon-design-system/carbon-addons-iot-react#3804]
### What package is this for?

- [X] React
- [ ] Angular

### Describe the bug

When IBM a11y scan is executed on storybook, violations are reported.

### To Reproduce

Run the accessibility scan in [storybook](http://localhost:9001/?path=/story/templates-%F0%9F%96%A5-tablisttemplate--tablisttemplate) and following are the violations
<img width="745" alt="image" src="https://github.com/carbon-design-system/carbon-addons-iot-react/assets/22093326/2b5dffc5-dc51-4212-9a41-8300768d9bca">

### Environment/versions:

- OS: MacOS
- Browser:chrome
- `carbon-addons-iot-react` version: [e.g. `v2.60.0`]
